### PR TITLE
Migrate files according to new convention

### LINKED ISSUE #251: Script to rename images and labels according to new convention
As mentioned in #88, we need a script to `rename images` in `data`, `text files` corresponding to `bug_id` and `label_files` along with `bounding_box_files` in `label_persons`

### PR DESCRIPTION
**Pull request by sagarvijaygupta**
Migrate data image files, text files, labelled files in label_persons according to the new conventions incorporated in #88

Fixes #251

### CODE REVIEW

**Comment by marco-c** [comment]
Let's put the code in a function `migrate_v1_to_v2`, as we might have other migrations in the future.
We should put a VERSION file in the data/ repository, so we know which migration should be applied.

**Comment by sagarvijaygupta** [comment]
What should we write in VERSION file?

**Comment by marco-c** [comment]
> What should we write in VERSION file?

Right now, just `1`. Then the migrate script should check the VERSION file, if it's `1` it should call `migrate_v1_to_v2` (which changes it to `2`).

**Comment by marco-c** [comment]
(The VERSION file should be added to the autowebcompat-data repo on GitLab)

**Comment by sagarvijaygupta** [comment]
> Then the migrate script should check the VERSION file, if it's 1 it should call migrate_v1_to_v2 (which changes it to 2)

You mean that after checking VERSION I should call `migrate_v1_to_v2` by default?
We expect future migrations of the form   `migrate_v2_to_v3`,  `migrate_3_to_v4`  or  `migrate_v1_to_v3`,  `migrate_v1_to_v4`?

**Comment by sagarvijaygupta** [comment]
@marco-c should I make the `to_version` as argparse input?

**Comment by marco-c** [comment]
> You mean that after checking VERSION I should call migrate_v1_to_v2 by default?
We expect future migrations of the form migrate_v2_to_v3, migrate_3_to_v4 or migrate_v1_to_v3, migrate_v1_to_v4?

`migrate_v1_to_v2` shouldn't always be called, but only if VERSION is == `1`.
Future migrations will be `migrate_v2_to_v3`, `migrate_v3_to_v4` and so on, this way we can support all possible migrations.

**Comment by marco-c** [comment]
> @marco-c should I make the to_version as argparse input?

The to_version should always be the latest (in this case 2).

**Comment by sagarvijaygupta** [comment]
Okay. So should I recursively call all of them and update the VERSION file in the end or we will just call it once and if wanted change VERSION file by ourselves and again call?

**Comment by marco-c** [comment]
> Okay. So should I recursively call all of them and update the VERSION file in the end or we will just call it once and if wanted change VERSION file by ourselves and again call?

Not recursively, but one after another. Each function should also update the VERSION file.

In pseudocode, something like this:
```
for i in range(CURRENT_VERSION, LATEST_VERSION - 1):
    migrate_v(i)_to_v(i+1)
```

**Comment by sagarvijaygupta** [comment]
@marco-c Can you see the VERSION file now? I added it previously also. I realized I had to commit in the submodule also before adding.

**Comment by marco-c** [comment]
You can't update the submodule without actually updating it on GitLab, otherwise it's not available to anyone else except you (infact, Travis fails to clone the submodule now, as the hash you provide is local to your machine).

Don't worry about adding the VERSION file, as I'm doing it now.

For the future, when you want to change something in data or tools, you actually have to send a pull request on GitLab, so that it can be merged and can be available to everybody.

**Comment by sagarvijaygupta** [comment]
Okay! I will try it next time. I actually wanted to update some new full_page_screenshots previously but was not able to at that time!

**Comment by sagarvijaygupta** [comment]
Should we keep a file along with VERSION file which corresponds to the screenshots present in that version. Since we won't be able to push images if generated from present crawler as after that migrate function will try to migrate them also.

**Comment by marco-c** [comment]
> Should we keep a file along with VERSION file which corresponds to the screenshots present in that version. Since we won't be able to push images if generated from present crawler as after that migrate function will try to migrate them also. (or simply the time stamp?)

I guess it's better if we just do the migration before adding new images.

File: migrate_files.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import os
+
+from autowebcompat import utils
+
+LATEST_VERSION = 5
```
Comment (marco-c) [inline migrate_files.py]: LATEST_VERSION should be 2

File: migrate_files.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        utils.write_bounding_boxes(new_bounding_boxes, os.path.join('label_persons', f))
+
+
+with open(os.path.join('data', 'VERSION'), 'r') as f:
+    CURRENT_VERSION = int(f.read())
+
+while CURRENT_VERSION != LATEST_VERSION:
+    migrate_function = globals()['migrate_v%d_to_v%d' % (CURRENT_VERSION, CURRENT_VERSION + 1)]
+    migrate_function()
+    CURRENT_VERSION += 1
```
Comment (marco-c) [inline migrate_files.py]: We also need to write the new version in the VERSION file

File: migrate_files.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import os
+
+from autowebcompat import utils
+
+LATEST_VERSION = 2
+
+
+def read_sequence(bug_id):
```
Comment (marco-c) [inline migrate_files.py]: This function and the next should be defined under `migrate_v1_to_v2`, as they are specific to this migration (if there's going to be a v2 to v3 in the future, these two functions will not be usable in both migrations).

File: migrate_files.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            continue
+        bug_id = os.path.splitext(f)[0]
+        data = read_sequence(int(bug_id))
+        write_sequence(int(bug_id), data)
+
+    for f in label_files:
+        if 'csv' not in f:
+            continue
+        labels = utils.read_labels(os.path.join('label_persons', f))
+        new_labels = {}
```
Comment (marco-c) [inline migrate_files.py]: You can directly modify `labels` instead.

File: migrate_files.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        for key, value in labels.items():
+            key_info = utils.parse_file_name(key)
+
+            if 'seq_no' not in key_info:
+                new_labels[key] = value
```
Comment (marco-c) [inline migrate_files.py]: if you directly modify labels instead, you won't need to do this

File: migrate_files.py (hunk 4 above)
Comment (marco-c) [reply migrate_files.py]: Actually, it's clearer this way, let's keep it this way.

File: migrate_files.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            browser = parts[2]
+            seq_no = map_names[seq_no]
+
+            key = utils.create_file_name(bug_id=bug_id, browser=browser, seq_no=seq_no)
+            new_bounding_boxes[key] = value
+        utils.write_bounding_boxes(new_bounding_boxes, os.path.join('label_persons', f))
+
+
+with open(os.path.join('data', 'VERSION'), 'r') as f:
+    CURRENT_VERSION = int(f.read())
```
Comment (marco-c) [inline migrate_files.py]: Nit: Usually upper case is for constants, let's use lower case here since you modify this later.

**Pull request merged**